spec/rspec/core/example_spec.rb

```diff
@@ -58,6 +58,21 @@ def metadata_hash(*args)
     end
   end
 
+  it "captures example timing even for exceptions unhandled by RSpec" do
+    begin
+      avoided = RSpec::Support::AllExceptionsExceptOnesWeMustNotRescue::AVOID_RESCUING
+      unhandled = Class.new(StandardError)
+      avoided << unhandled
```

Comment:
I considered it, but it felt really gross with any of the exceptions in that collection. Maybe OutOfMemory is the least dangerous one?

Bradley Schaefer

> On Dec 30, 2015, at 12:28 PM, Myron Marston <notifications@github.com> wrote:
> 
> In spec/rspec/core/example_spec.rb:
> 
> > @@ -58,6 +58,21 @@ def metadata_hash(*args)
> >      end
> >    end
> >  
> > +  it "captures example timing even for exceptions unhandled by RSpec" do
> > +    begin
> > +      avoided = RSpec::Support::AllExceptionsExceptOnesWeMustNotRescue::AVOID_RESCUING
> > +      unhandled = Class.new(StandardError)
> > +      avoided << unhandled
> Mutating a global constant feels kinda gross. Can we just use one of the exceptions already in it instead of adding a new one to it?
> 
> —
> Reply to this email directly or view it on GitHub.
>